src/portfft/dispatcher/subgroup_dispatcher.hpp

```diff
@@ -256,7 +263,13 @@ PORTFFT_INLINE void subgroup_impl(const T* input, T* output, const T* input_imag
             }
           }
         }
-        sg_dft<Dir, SubgroupSize>(priv, global_data.sg, factor_wi, factor_sg, loc_twiddles, wi_private_scratch);
+        if (take_conjugate_on_load) {
+          take_conjugate(priv, factor_wi);
+        }
+        sg_dft<SubgroupSize>(priv, global_data.sg, factor_wi, factor_sg, loc_twiddles, wi_private_scratch);
+        if (take_conjugate_on_store) {
+          take_conjugate(priv, factor_wi);
+        }
```

Comment:
We do this kind of thing sufficeintly often that we might consider a wrapper function.

Reply:
you mean just for the subgroup one, or all three ?

Reply:
In general, every time we have the pattern
```        
        if (take_conjugate_on_load) {
          take_conjugate(priv, factor_wi);
        }
        XXX_dft<SubgroupSize>(priv, global_data.sg, factor_wi, factor_sg, loc_twiddles, wi_private_scratch);
        if (take_conjugate_on_store) {
          take_conjugate(priv, factor_wi);
        }
```
it could be wrapped as
```
        XXX_dft<SubgroupSize>(priv, global_data.sg, factor_wi, factor_sg, loc_twiddles, wi_private_scratch, conj_on_load, conj_on_store);
```

Whilst I'd be tempted to suggest an elegant template-based solution ™️ it'd probably be horrible for compiler error messages. Do what you think best (including deciding this is best as is), and resolve this comment.

Reply:
I would avoid this at the moment, even though it's a nice suggestion because this can also be extended to our application of scale factor and modifiers as well, as, in our implementations, it's the following steps, in order on the `priv` array

* conjugate_on_load
* load_modifier
* XXX_DFT
* conjugate_on_store
* store_modifier
* apply_scale_factor

also given that different implementations take different parameters, it might lead to specialized functions per implementations, something which I would like to avoid, as the point of having such a function would be to provide a common interface for the above order of operations across all the three implementations, hence best left as a separate PR which would come up with a well thought implementation of the same.